tests/unit/Referrals/ReferralDetectorTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace MailPoet\Referrals;
+
+use MailPoet\Settings\SettingsController;
+use MailPoet\WP\Functions as WPFunctions;
+use PHPUnit_Framework_MockObject_MockObject as MockObject;
+
+class ReferralDetectorTest extends \MailPoetUnitTest {
+
+  /** @var MockObject */
+  private $settings_mock;
+
+  /** @var MockObject */
+  private $wp_mock;
+
+  function _before() {
+    $this->settings_mock = $this->createMock(SettingsController::class);
+    $this->wp_mock = $this->createMock(WPFunctions::class);
+    if (!defined(ReferralDetector::REFERRAL_CONSTANT_NAME)) {
+      define(ReferralDetector::REFERRAL_CONSTANT_NAME, 'constant_referral_id');
+    }
+  }
+
+  function testItPrefersSettingsValueOverAll() {
```

Comment:
The method name sounds weird to me. Probably `testItPrefersSettingsValue` is enough but "others" or "any other" would be correct, same as "prefer to" instead of "prefer over".